src/units/abilities.cpp

```diff
@@ -1189,7 +1190,10 @@ static std::vector<std::string> damage_type_list(const unit_ability_list& abil_l
 			});
 		}
 	}
-	return type_list;
+	if(!type_list.empty()){
```

Comment:
No need to declare the temp variable at the top of the function. Just simplify this check:
```cpp
if(!type_list.empty()) {
	return type_list.front();
} else {
	return std::nullopt;
}
```

Reply:
done